### PR TITLE
Restructuring sample

### LINKED ISSUE #201: Generate unique 10-characters ids for samples
https://www.npmjs.com/package/crypto-random-string

### LINKED ISSUE #203: How to have hierarchy in samples ?
sampleId is currently an array of string

It seems useful that if a sample is a child of another sample we just append a string to the previous array.

This will allow to display the hierarchy of samples.

### PR DESCRIPTION
**Pull request by maasencioh**
closes #201 #203

### CODE REVIEW

File: app/Controllers/Http/MeasurementController.ts [hunk 1]
```diff
@@ -70,20 +70,13 @@ export default class MeasurementController {
...
-        });
+        sample.attachments.push({ id: fileId, date: new Date() });
       }
-      const measurementId = measurement._id.toHexString();
-      sample.activities.push({
-        type: ActivityType.MEASUREMENT,
-        measurementId: measurementId,
-        measurementType: collection,
+      sample.measurements.push({
+        id: measurement._id.toHexString(),
```
Comment (targos) [inline app/Controllers/Http/MeasurementController.ts]: why not keep the ObjectID?

File: app/Resolvers/Sample.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      const ans = await SampleKind.find(id);
+      if (ans) return ans.toJSON() as GqlSampleKind;
+
+      throw new UserInputError('Id not found', { argumentName: 'id' });
+    },
+  },
+  Mutation: {
+    async createSample(_, { input }) {
+      const sample = await Sample.fromInput(new Sample(), input);
+      return sample.toJSON() as Sample;
```
Comment (targos) [inline app/Resolvers/Sample.ts]: why do you have to call `toJSON()` ?

File: app/Resolvers/Sample.ts [hunk 3]
```diff
@@ -70,9 +69,12 @@ const resolvers: GqlResolvers = {
       return parent;
     },
     async children(sample: Sample) {
-      return Sample.query({ 'parents.0': sample._id }).all();
+      return Sample.query({ 'parents.0': sample._id.toHexString() }).all();
```
Comment (targos) [inline app/Resolvers/Sample.ts]: something's wrong. references to ObjectIDs should be ObjectIDs in the database, not strings.

File: app/Resolvers/Sample.ts (hunk 2 above)
Comment (maasencioh) [reply app/Resolvers/Sample.ts]: it was a wrong assumption that I had

**Review by targos — APPROVED** [review]

**Pull request merged**